api/app/src/test/kotlin/packit/unit/controllers/PacketControllerTest.kt

```diff
@@ -76,22 +78,30 @@ class PacketControllerTest
         )
     )
 
-    private val htmlContentByteArray = "<html><body><h1>Test html file</h1></body></html>".toByteArray()
-
-    private val inputStream = ByteArrayResource(htmlContentByteArray) to HttpHeaders.EMPTY
-
     private val mockPageablePackets = PageImpl(packets)
-
+    private val testHeaders = HttpHeaders().apply { contentLength = 100L }
     private val packetService = mock<PacketService> {
         on { getPackets(PageablePayload(0, 10), "", "") } doReturn mockPageablePackets
-        on { getFileByHash(anyString(), anyBoolean(), anyString()) } doReturn inputStream
+        on {
+            getFileByHash(
+            anyString(),
+            any<OutputStream>(),
+            any<(HttpHeaders) -> Unit>()
+        )
+        } doAnswer { invocationOnMock ->
+            val outputStream = invocationOnMock.getArgument<OutputStream>(1)
+            outputStream.write("mocked output content".toByteArray())
+            val callback = invocationOnMock.getArgument<(HttpHeaders) -> Unit>(2)
+            callback(testHeaders)
+        }
         on { getPacketsByName(anyString()) } doReturn packets
 
         packetMetadata.forEach {
             on { getMetadataBy(it.id) } doReturn it
         }
     }
 
+    private val response = MockHttpServletResponse()
```

Comment:
Doesn't this making a global value cause problems with it being shared across tests. For example on test runs, passes it to a controller method that calls `setHeader` on it, the next test runs and it inherits sees the header. eg. does the response in the `streamZip` test have a left over content length from the `streamFile` tests?

I think it would be safer to create one in each test function.

(Usually we get away with global mocks and SUTs because they are immutable objects, but this is not)